aten/src/ATen/native/cuda/Loss.cu

```diff
@@ -445,17 +443,18 @@ __global__ void nll_loss_backward_reduce_cuda_kernel_2d(
     bool size_average,
     int nframe,
     int ndim,
-    int n_classes,
+    int64_t n_classes,
     int64_t ignore_index) {
   const auto grad = -(size_average ? *grad_output / *total_weight
                                    : *grad_output);
 
   for (int i = threadIdx.x; i < nframe; i += NLL_LOSS_THREADS) {
-    int t = target[i];
-    if (t != static_cast<int>(ignore_index)) {
+    const int64_t t = static_cast<int64_t>(target[i]);
```

Comment:
There's no real need to make the cast explicit (same on the other functions), but whatever works.